### ISSUE TITLE
Add keys to limit access to the API

### ISSUE THREAD

**Issue by michaeldera** [michaeldera/image-accessibility-extension#5]
**Is your feature request related to a problem? Please describe.**
At the moment the api is open but it will need to be closed if there is mass usage in case the endpoint to the api is used for purposes other than to support this extension. So the extension must have a key it uses to authenticate with the API to be granted access.

**Describe the solution you'd like**
A key tied to the extension (preferred) or accounts for the users 

**Describe alternatives you've considered**
Rate-limiting from the API end.

**Comment by nadchif** [comment]
May I suggest both
 1. Require an account 
2. Apply rate-limiting to target fake or bot accounts

See also: https://developer.chrome.com/webstore/identify_user

**Comment by michaeldera** [comment]
Rate limiting would need to be implemented in the backend... So I will start work on that.

So for the accounts we just do the Google accounts or Also have the option for Microsoft Accounts (Considering that this will also be on Edge).... but naturally we would go for Google first?

**Comment by nadchif** [comment]
Agree, I Microsoft Accounts would also be very convenient for Edge users, but we can start off with the Google Accounts on Chrome

**Comment by michaeldera** [comment]
Agreed... We can start with  Chrome.

**Comment by michaeldera** [comment]
@nadchif do you want to take this or should I work on it?

**Comment by nadchif** [comment]
I'd love to do it, but I may need to constantly communicate with you with regards the the backend side of this. 

Some "real-time" method of communication would be convenient. Are you open to using slack?

**Comment by michaeldera** [comment]
Sure.. We can do that. The backend repo is also opn here. We can use Slack.
